lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -156,6 +159,21 @@ def __init__(self, charm: CharmBase) -> None:
             charm.on[self.relation_name].relation_broken, self._on_relation_broken
         )
 
+    @property
+    def dashboards_relations(self):
+        """Return the dashboard relations out of all."""
+        result = []
+        for relation in self.opensearch_provides.relations:
+            # Won't work... There's NO trace of any relation details that would
+            # allow for decision-making here :-(
+            #
+            # if KibanaserverRole in self.opensearch_provides.fetch_relation_field(
+            #     relation.id, "extra_user_roles"
+            # ):
+            if any(key.name == "opensearch-dashboards" for key in relation.data.keys()):
+                result.append(relation)
```

Comment:
...is this really what we want...?

If yes, I move the sting to a constant. If not... I'm running out of ideas :-(

Reply:
Wouldn't what you commented out work here?
```
if KibanaserverRole in self.opensearch_provides.fetch_relation_field(
    relation.id, "extra_user_roles"
)
```
I prefer it over the `key.name` approach

Reply:
I added 

```
            # Won't work... There's NO trace of any relation details that would
            # allow for decision-making here :-(
```
comment above the commented code to highlight that it wouldn't work, and why.

Reply:
Sorry, I could be wrong, but I think the below works:
```
dashboards_relations = []
for relation in self.model.relations[self.relation_name]:
    remote_data = relation.data[relation.app]
    if KibanaServerRole in remote_data.get("extra_user_roles", []):
         dashboard_relations.append(relation.id)
```

Reply:
https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/opensearch_client/v0#example

Reply:
Let me double-check in practice if I overlooked something.

I couldn't find the field in the databag.

Reply:
All right my bad.

I don't know what I must have overlooked.

But the field is there indeed

```
  - relation-id: 5
    endpoint: opensearch-client
    related-endpoint: opensearch_client
    application-data:
      extra-user-roles: kibanaserver
      index: .opensearch-dashboards
      requested-secrets: '["username", "password", "tls", "tls-ca", "uris"]'
    related-units:
      opensearch-dashboards/0:
        in-scope: true
        data:
          data: '{"endpoints": "10.196.115.48:9200", "index": ".opensearch-dashboards",
            "secret-tls": "secret://4a938601-7354-49d8-8f2a-e0f434d2ea48/coc18l3v6d6i9akiaqd0",
            "secret-user": "secret://4a938601-7354-49d8-8f2a-e0f434d2ea48/coc18l3v6d6i9akiaqcg",
            "version": "2.12.0"}'
          egress-subnets: 10.196.115.189/32
          ingress-address: 10.196.115.189
          private-address: 10.196.115.189
```